src/server/db_slice.cc

```diff
@@ -453,7 +453,7 @@ OpResult<DbSlice::PrimeItAndExp> DbSlice::FindInternal(const Context& cntx, std:
   }
 
   if (res.it->second.HasExpire()) {  // check expiry state
-    res = ExpireIfNeeded(cntx, res.it, true);
+    res = ExpireIfNeeded(cntx, res.it, false);
```

Comment:
when  we call FindInternal we always hold the lock to the key, or at least we should, if not its probably a bug

Reply:
yes and I think I am wrong here as well. There is something else that is happening.